crates/block-producer/src/runner.rs

```diff
@@ -875,13 +888,45 @@ pub async fn run(config: Config, skip_config_check: bool) -> Result<()> {
                         shutdown_send,
                         shutdown_event_recv,
                     );
-                    while let Err(err) = chain_task.run(&mut backoff).await {
-                        if err.is::<RPCRequestError>() {
-                            log::error!("chain polling loop request error, will retry: {}", err);
-                            tokio::time::sleep(backoff.next_sleep()).await;
-                        } else {
-                            log::error!("chain polling loop exit unexpected, error: {}", err);
-                            break;
+
+                    let mut run_status = ChainTaskRunStatus::default();
+                    loop {
+                        // Exit if shutdown event is received.
+                        if chain_task.shutdown_event.try_recv().is_ok() {
+                            log::info!("ChainTask existed successfully");
+                            return;
+                        }
+
+                        let chain_task_run_span = info_span!("chain_task_run");
+                        let _run_guard = chain_task_run_span.enter();
+
+                        match chain_task.run(&run_status).await {
```

Comment:
Attach `chain_task_run_span` to `chain_task.run(..)`?

E.g. `chain_task.run(&run_status).instrument(chain_task_run_span).await`.

See: https://docs.rs/tracing/latest/tracing/struct.Span.html#in-asynchronous-code

Reply:
Fixed

Reply:
On another look, `run` already has an `instrument` attribute, so this shouldn't be necessary?